CONTRIBUTING.md

```diff
@@ -0,0 +1,158 @@
+We love your enthusiasm, but small changes and small PRs are easier to digest. We're all doing this in our spare time, it is easier to review a couple of small things and merge that and iterate from there than to have a PR with 100+ files changed that will sit there forever.
+
+#### Added features should have tests, a sample and documentation
+We like quality as much as the next person, so please provide tests.
+
+In addition, we would want a new feature or change to be as clear as possible for other developers. Please add a sample to the sample app as part of your PR and also provide a PR to our [Add link to Microsoft Docs GitHub Repo Here]
+
+## Contributing Code - Best Practices
+
+### Enums
+* Always use `Unknown` at index 0 for return types that may have a value that is not known
+* Always use `Default` at index 0 for option types that can use the system default option
+* Follow naming guidelines for tense... `SensorSpeed` not `SensorSpeeds`
+* Assign values (0,1,2,3) for all enums
+
+### Property Names
+* Include units only if one of the platforms includes it in their implementation. For instance HeadingMagneticNorth implies degrees on all platforms, but PressureInHectopascals is needed since platforms don't provide a consistent API for this.
+
+### Units
+* Use the standard units and most well accepted units when possible. For instance Hectopascals are used on UWP/Android and iOS uses Kilopascals so we have chosen Hectopascals.
+
+### Pattern matching
+
+#### Null checking
+* Prefer using `is` when checking for null instead of `==`.
+
+e.g. 
+
+```csharp
+// null
+if (something is null)
+{
+
+}
+
+// or not null
+if (something is not null)
+{
+   
+}
+```
+
+* Avoid using the `!` [null forgiving operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving) to avoid the unintended introduction of bugs.
+
+#### Type checking
+
+* Prefer `is` when checking for types instead of casting.
+
+e.g.
+
+```csharp
+if (something is Bucket bucket)
+{
+   bucket.Empty();
+}
+```
+
+### File Scoped Namespaces
+* Use [file scoped namespaces](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces) to help reduce code verbosity.
+
+e.g. 
+
+```csharp
+namespace CommunityToolkit.Maui.Converters;
+
+using System;
+
+class BoolToObjectConverter
+{
+}
+```
+
+### Bug Fixes
+
+If you're looking for something to fix, please browse [open issues](https://github.com/CommunityToolkit/Maui/issues). 
+
+Follow the style used by the [.NET Foundation](https://github.com/dotnet/runtime/blob/master/docs/coding-guidelines/coding-style.md), with two primary exceptions:
+
+- We do not use the `private` keyword as it is the default accessibility level in C#.
+- We will **not** use `_` or `s_` as a prefix for internal or private field names
+- We will use `camelCaseFieldName` for naming internal or private fields in both instance and static implementations
```

Comment:
This is from XCT but it appears we have a inconsistent approach already in MCT:

* `_` prefix -> https://github.com/CommunityToolkit/Maui/blob/00f8f8de9e81423a43800404c1d6ba932f8a7690/src/CommunityToolkit.Maui/Alerts/Snackbar/SnackBar.shared.cs#L11
* `camelCaseFieldName` -> https://github.com/CommunityToolkit/Maui/blob/00f8f8de9e81423a43800404c1d6ba932f8a7690/src/CommunityToolkit.Maui/Behaviors/BaseBehavior.shared.cs#L21

Reply:
can we fix it in https://github.com/CommunityToolkit/Maui/pull/214 or it will be a new PR with editorConfig rules?

Reply:
I don't mind. If #214 is near completion I'm happy to include in this or whichever PR includes the .editorconfig addition.

I guess we should decide which approach we prefer first 😃

Reply:
Let's make those changes here in this PR.

#214 is a massive PR and just got 2 Approvals. I'm exhausted just thinking about reviewing it again 😅

Reply:
For what it's worth, I prefer prefixing fields with `_`.

That's just my personal preference (makes it easy to determine at a glance whether a variable's scope is local or global). It's not a hill I'm willing to die on, though, if everyone else prefers to nix the `_` prefix.

Reply:
I lean the other way but again I'm not going to fight to the death on it. If it means we have fewer differences to the linked standard from dotnet foundation then that's probably a better approach. To clarify this would be in favour of the `_` prefix